app/services/recent-events.ts

```diff
@@ -156,19 +223,43 @@ export class RecentEventsService extends StatefulService<IRecentEventsState> {
           'displayString',
           'comment',
           'title',
+          'read',
+          'hash',
         ]);
       });
 
       eventArray = eventArray.concat(culledEvents);
     });
 
+    // Format string of keys to look for in server event cache
+    const hashValues = eventArray.map(event => event.hash).join('|##|');
```

Comment:
This is bizarre

Reply:
Yep, this API is particularly unwieldy for no discernible reason